src/c++/perf_analyzer/test_request_rate_manager.cc

```diff
@@ -429,44 +398,44 @@ TEST_CASE("request_rate_sequence")
   trrm.TestSequences();
 }
 
-TEST_CASE("request_rate_streaming: test that streaming-specific logic works")
-{
-  PerfAnalyzerParameters params{};
-  params.streaming = true;
-
-  std::shared_ptr<TestRequestRateManager::ThreadStat> thread_stat{
-      std::make_shared<TestRequestRateManager::ThreadStat>()};
-  std::shared_ptr<TestRequestRateManager::ThreadConfig> thread_config{
-      std::make_shared<TestRequestRateManager::ThreadConfig>(0, 0)};
-
-  SUBCASE("enable_stats true")
-  {
-    TestRequestRateManager trrm(params);
-    trrm.schedule_.push_back(std::chrono::nanoseconds(1));
-
-    std::future<void> infer_future{std::async(
-        &TestRequestRateManager::Infer, &trrm, thread_stat, thread_config)};
-
-    early_exit = true;
-    infer_future.get();
-
-    CHECK(trrm.stats_->start_stream_enable_stats_value == true);
-  }
-
-  SUBCASE("enable_stats false")
-  {
-    TestRequestRateManager trrm(
-        params, false /* is_sequence */, true /* is_decoupled */);
-    trrm.schedule_.push_back(std::chrono::nanoseconds(1));
-
-    std::future<void> infer_future{std::async(
-        &TestRequestRateManager::Infer, &trrm, thread_stat, thread_config)};
-
-    early_exit = true;
-    infer_future.get();
-
-    CHECK(trrm.stats_->start_stream_enable_stats_value == false);
-  }
-}
-
+// FIXME how to revive these with Infer gone?
```

Comment:
I need to revive this test

Reply:
I think the way would be to create like a TestRequestRateWorker class that basically does the same stuff as this test. Instantiates a TestRequestRateWorker object and calls the infer function. There's a loooooooooooooooooot of parameters for the RequestRateWorker constructor, unfortunately, which will make writing these tests difficult.

Reply:
I created an interface, which among other things allows me to get around needing all of those parameters. All tests are now revived.